File: src/main/java/org/jukeboxmc/network/handler/InteractHandler.java
package org.jukeboxmc.network.handler;

import org.cloudburstmc.protocol.bedrock.packet.InteractPacket;
import org.jukeboxmc.Server;
import org.jukeboxmc.player.Player;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class InteractHandler implements PacketHandler<InteractPacket> {
    @Override
    public void handle( InteractPacket packet, Server server, Player player ) {
        if ( packet.getAction().equals( InteractPacket.Action.OPEN_INVENTORY ) ) {
            player.openInventory( player.getInventory() );
        }
    }
}

File: JukeboxMC-Server/src/main/kotlin/org/jukeboxmc/server/network/handler/InteractHandler.kt
package org.jukeboxmc.server.network.handler

import org.cloudburstmc.protocol.bedrock.packet.InteractPacket
import org.jukeboxmc.server.JukeboxServer
import org.jukeboxmc.server.player.JukeboxPlayer

class InteractHandler : PacketHandler<InteractPacket> {

    override fun handle(packet: InteractPacket, server: JukeboxServer, player: JukeboxPlayer) {
       if (packet.action == InteractPacket.Action.OPEN_INVENTORY) {
           player.openInventory(player.getInventory())
       }
    }
}